### DISCUSSION TITLE
Track style changes

### DISCUSSION THREAD

**Discussion by wvbe** [dolanmiu/docx#1586] [Ideas / Requests]
MS Word has this feature that change tracking can show which (paragraph) styles have changed. In the project that I'm using `docx` for this is a significant change, sometimes even more important than just the text change.

![image](https://user-images.githubusercontent.com/4198877/178293872-51794421-a980-46c9-909b-409eaabc57cf.png)

Because `Paragraph` does not have a `revision` option, and `revision.style` on `TextRun` does not appear to have an effect, I'm assuming that creating a document where the paragraph style change is recorded is currently not a feature.

I'm wondering if somebody could give me a few pointers wrt. how I could implement this in such a way that a PR is likely to make it back into `docx`!

----

ps. the `document.xml` snippet belonging to above screenshot is;

```xml
<w:p w14:paraId="1852C87C" w14:textId="6AD2A34B" w:rsidR="00C8746D" w:rsidRPr="00A44254" w:rsidRDefault="00A44254" w:rsidP="00A44254">
	<w:pPr>
		<w:pStyle w:val="Heading2" />
		<w:rPr>
			<w:lang w:val="en-US" />
		</w:rPr>
		<w:pPrChange w:id="2" w:author="Wybe Minnebo" w:date="2022-07-11T16:56:00Z">
			<w:pPr>
				<w:pStyle w:val="Heading1" />
			</w:pPr>
		</w:pPrChange>
	</w:pPr>
	<w:r w:rsidRPr="00A44254">
		<w:rPr>
			<w:lang w:val="en-US" />
		</w:rPr>
		<w:lastRenderedPageBreak />
		<w:t>This used to be a “Heading 1”</w:t>
	</w:r>
</w:p>
```

**Comment by minyoung90** [comment]
I need help too!
